Formula/objc-run.rb

```diff
@@ -6,13 +6,18 @@ class ObjcRun < Formula
   license "MIT"
   head "https://github.com/iljaiwas/objc-run.git"
 
-  bottle :unneeded
-
   def install
+    # Keep bottles uniform before keg-relocation
+    inreplace "objc-run", "/usr/local", HOMEBREW_PREFIX
     bin.install "objc-run"
     pkgshare.install "examples", "test.bash"
   end
 
+  def post_install
+    # Undo mistaken keg relocation
+    inreplace bin/"objc-run", HOMEBREW_PREFIX, "/usr/local"
+  end
```

Comment:
Thanks for doing this but I think it's the wrong approach. I don't think it's ever worth undoing a change in `post_install` even if it gets us an `:all` bottle. In this case it's the matching of `$projectDirPath/build/usr/local/bin/clitest` as `HOMEBREW_PREFIX` that's wrong so we should fix the relocation accordingly in Homebrew/brew.

Reply:
Yes, I'm not stoked about the `post_install` method, and was contemplating fixes in Homebrew/brew: hence the suggestion of `skip_relocation` in #75943.

We could tighten the `HOMEBREW_PREFIX` matching to get rid of false positives like this, but I'm worried about false negatives (paths that should match, and do match currently, but won't under a potential tightening of `HOMEBREW_PREFIX` matching).

It seems to me that it is much safer to just identify certain formula as ones where `brew` should skip all relocation entirely. This also has the benefit of rendering all further `inreplace`s to get `:all` bottles unnecessary.

Reply:
> We could tighten the `HOMEBREW_PREFIX` matching to get rid of false positives like this, but I'm worried about creating more false negatives (paths that should match, and do match currently, but won't under a potential tightening of `HOMEBREW_PREFIX` matching).

The case that seems reasonable to me is a match like `/[^a-z]?#{HOMEBREW_PREFIX}/` rather than `/#{HOMEBREW_PREFIX}/`

> It seems to me that it is much safer to just identify certain formula as ones where `brew` should skip all relocation entirely. This also has the benefit of rendering all further `inreplace`s to get `:all` bottles unnecessary.

The issue to me here is: how do you know that all relocation should be skipped? How do you know this is still the case for every upgraded version?

Reply:
> The case that seems reasonable to me is a match like `/[^a-z]?#{HOMEBREW_PREFIX}/` rather than `/#{HOMEBREW_PREFIX}/`

That is certainly reasonable. However, I'm not sure I am comfortable with the assumption that all parsers (at least, the ones used by any formulae we package) are reasonable. I will admit to not having any problematic examples in mind, so I am likely just overly paranoid here.

> The issue to me here is: how do you know that all relocation should be skipped?

My understanding was that anything that is/was `bottle :unneeded` (and would've been were we not bottling everything) should not need any relocation. Am I mistaken here?

> How do you know this is still the case for every upgraded version?

The same way we tell that a formula is still `bottle :unneeded` after it is upgraded. (Of course, this would be wrong if it isn't the case that `bottle :unneeded` formulae actually don't need relocation.)

Reply:
> That is certainly reasonable. However, I'm not sure I am comfortable with the assumption that all parsers (at least, the ones used by any formulae we package) are reasonable. I will admit to not having any problematic examples in mind, so I am likely just overly paranoid here.

Yes, with love, I think you're being overly paranoid 😁 

> My understanding was that anything that is/was `bottle :unneeded` (and would've been were we not bottling everything) should not need any relocation. Am I mistaken here?

They shouldn't but already we've found some that were.

> The same way we tell that a formula is still `bottle :unneeded` after it is upgraded. (Of course, this would be wrong if it isn't the case that `bottle :unneeded` formulae actually don't need relocation.)

Same deal as above! Some were missed in this case when they shouldn't have been.

Reply:
Fair enough! I'll see if I can look into tightening `HOMEBREW_PREFIX` matching in keg-relocation. I've had some other thoughts related to it anyway, so I've been meaning to poke at that bit of `brew` already.

Reply:
@carlocab Thanks but no need to do if you don't want to! I think it's worth reverting this change for now unless it's known to be broken without it.

Reply:
Do you mean making it `bottle :unneeded` again, or just removing the `post_install` method?

Reply:
@carlocab Keep it bottled, remove both `inreplace`s, don't care about it not being `all:` (for now)

Reply:
Not sure that's the right thing to do here, but I'll open a PR doing that where we can continue the discussion/give others the chance to weigh in.